### PR TITLE
Migrate to v2 of the intellij gradle plugin

### PR DESCRIPTION
**Pull request by nrayburn-tech**
- Migrate to v2 of the IntelliJ gradle plugin (Mostly just syncing with the IntelliJ plugin template repo)
- Update plugin name to fix CI errors during the verify action
- Update target IDE versions due to incompatibilities with the LSP classes between different IDE versions

This will close out most (all?) of the pending dependabot PRs.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,34 +1,53 @@
...
 
 group = properties("pluginGroup").get()
 version = properties("pluginVersion").get()
 
 // Configure project's dependencies
 repositories {
-    maven(url = "https://maven.aliyun.com/repository/central")
-    maven(url = "https://maven.aliyun.com/repository/public")
-    maven(url = "https://maven.aliyun.com/repository/gradle-plugin")
-    maven(url = "https://maven.aliyun.com/repository/gradle-core")
```
Comment (nrayburn-tech) [inline build.gradle.kts:23]: If these URLs are needed for some reason then I can probably figure out how to retain them with the new changes.

**Comment by nrayburn-tech** [comment]
The verify plugin build is failing due to the name of the plugin in `src/main/resources/META-INF/plugin.xml`.  I don't think the name is supposed to contain `intellij` or `plugin`, but since it was existing I didn't want to change it.

**Review by IWANABETHATGUY — APPROVED** [review]

**Comment by IWANABETHATGUY** [comment]
Ci failed

**Comment by nrayburn-tech** [comment]
@IWANABETHATGUY it's going to fail until https://github.com/oxc-project/oxc-intellij-plugin/pull/50#issuecomment-2401310949 is resolved.  If you have a plugin name suggestion, I can change it.  Maybe even just `oxc` is fine.

**Comment by nrayburn-tech** [comment]
Running verify locally and found some compatibility issues with the existing LSP code and the targeted IDE versions as well.  Went ahead and renamed the plugin and fixed these.
- Changed plugin name to `Oxc`.
- Changed the target IDE version to only 2023 or 2024 versions.

**Comment by IWANABETHATGUY** [comment]
> The verify plugin build is failing due to the name of the plugin in `src/main/resources/META-INF/plugin.xml`. I don't think the name is supposed to contain `intellij` or `plugin`, but since it was existing I didn't want to change it.

Still not sure why the ci is failed, since it is sucessed before? Only because of the naming?

**Comment by nrayburn-tech** [comment]
Running the main branch with the old command `runPluginVerifier` and it doesn't fail locally, it only emits a warning (same as CI).  My best guess is that the newer plugin and task is failing now instead of just showing a warning.

**Comment by IWANABETHATGUY** [comment]
> Running verify locally and found some compatibility issues with the existing LSP code and the targeted IDE versions as well. Went ahead and renamed the plugin and fixed these.
> 
> * Changed plugin name to `Oxc`.
> * Changed the target IDE version to only 2023 or 2024 versions.

@Boshen ,Can you give a better naming?

**Comment by IWANABETHATGUY** [comment]
https://github.com/biomejs/biome-intellij/blob/main/src/main/resources/META-INF/plugin.xml, I found biome just use `Biome`  I think use `Oxc` is good enough.

**Pull request merged**